src/compiler/checker.ts

```diff
@@ -49508,12 +49506,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function checkExternalEmitHelpers(location: Node, helpers: ExternalEmitHelpers) {
-        if ((requestedExternalEmitHelpers & helpers) !== helpers && compilerOptions.importHelpers) {
-            const sourceFile = getSourceFileOfNode(location);
-            if (isEffectiveExternalModule(sourceFile, compilerOptions) && !(location.flags & NodeFlags.Ambient)) {
+        const sourceFile = getSourceFileOfNode(location);

```

Comment:
Could we preserve the `if (compilerOptions.importHelpers)` condition here? By moving the check to line 49513 you're doing a lot of extra work, including needlessly increasing the size of the node links table and defining the `requestedExternalEmitHelpers` property.

Reply:
I still think this should move up before calling `getSourceFileOfNode` to avoid the unnecessary walk up the spine.

Reply:
Done.